File: ios/KPOPVOTE/KPOPVOTE/Models/ExternalAppMaster.swift
//
//  ExternalAppMaster.swift
//  KPOPVOTE
//
//  K-VOTE COLLECTOR - External App Master Model
//

import Foundation

struct ExternalAppMaster: Identifiable, Codable {
    let id: String
    let appName: String
    let appUrl: String
    let iconUrl: String?
    let createdAt: Date?
    let updatedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id = "appId"
        case appName
        case appUrl
        case iconUrl
        case createdAt
        case updatedAt
    }

    // Computed property for display
    var displayName: String {
        appName
    }
}

// MARK: - Sample Data for Preview
extension ExternalAppMaster {
    static let sampleData: [ExternalAppMaster] = [
        ExternalAppMaster(
            id: "idol-champ",
            appName: "IDOL CHAMP",
            appUrl: "https://www.idolchamp.com",
            iconUrl: "https://firebasestorage.googleapis.com/v0/b/kpopvote-9de2b.appspot.com/o/app-icons%2Fidol_champ.png?alt=media",
            createdAt: Date(),
            updatedAt: Date()
        ),
        ExternalAppMaster(
            id: "mnet-plus",
            appName: "Mnet Plus",
            appUrl: "https://www.mnetplus.world",
            iconUrl: "https://firebasestorage.googleapis.com/v0/b/kpopvote-9de2b.appspot.com/o/app-icons%2Fmnet_plus.png?alt=media",
            createdAt: Date(),
            updatedAt: Date()
        ),
        ExternalAppMaster(
            id: "mubeat",
            appName: "MUBEAT",
            appUrl: "https://www.mubeat.io",
            iconUrl: "https://firebasestorage.googleapis.com/v0/b/kpopvote-9de2b.appspot.com/o/app-icons%2Fmubeat.png?alt=media",
            createdAt: Date(),
            updatedAt: Date()
        )
    ]
}

File: android/app/src/main/kotlin/com/kpopvote/collector/data/model/ExternalAppMaster.kt
package com.kpopvote.collector.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Mirrors iOS `ExternalAppMaster.swift`. */
@Serializable
data class ExternalAppMaster(
    @SerialName("appId") val id: String,
    val appName: String,
    val appUrl: String,
    val iconUrl: String? = null,
    val defaultCoverImageUrl: String? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null,
) {
    val displayName: String get() = appName
}

@Serializable
data class ExternalAppListData(
    val apps: List<ExternalAppMaster>,
    val count: Int = 0,
)
